static/js/browser2/observation_chart.tsx

```diff
@@ -55,44 +55,111 @@ export class ObservationChart extends React.Component<
   constructor(props: ObservationChartPropType) {
     super(props);
     this.state = {
-      canClickDots: false,
       dateToDcid: {},
       errorMessage: "",
+      showTableView: false,
     };
+    this.getObsDcidRequest = this.getObsDcidRequest.bind(this);
   }
 
   componentDidMount(): void {
     this.plot();
-    if (this.props.hasClickableDots) {
+    if (this.props.canClickObs) {
       this.fetchObservationDcidsData();
     }
   }
 
   render(): JSX.Element {
-    let svgContainerClass = this.state.canClickDots ? "clickable" : "no-click";
+    let svgContainerClass = this.props.canClickObs ? "clickable" : "no-click";
     if (Object.keys(this.props.sourceSeries.val).length > MAX_DOTS) {
       svgContainerClass = svgContainerClass + " hide-dots";
     }
+    const obsTableRowClass = this.props.canClickObs
+      ? "observation-table-row-clickable"
+      : "observation-table-row";
+    const chartVisibility = this.state.showTableView ? "none" : "block";
+    const tableVisibility = this.state.showTableView ? "block" : "none";
+    const unit = getUnit(this.props.sourceSeries);
     return (
       <>
-        <div id={this.svgContainerId()} className={svgContainerClass} />
+        <div>
+          <i className="material-icons">
+            {this.state.showTableView ? "show_chart" : "table_view"}
+          </i>
+          <span
+            className="clickable-text"
+            onClick={() =>
+              this.setState({ showTableView: !this.state.showTableView })
+            }
+          >
+            {this.state.showTableView ? "show chart" : "show table"}
+          </span>
+        </div>
+        <div
+          id={this.svgContainerId() + "-content-area"}
+          style={{ position: "relative" }}
+        >
+          <div style={{ display: tableVisibility }}>
+            <div className="observations-table">
+              <table className="node-table">
+                <tbody>
+                  <tr key="header">
+                    <td width="50%">
+                      <strong>Date</strong>
+                    </td>
+                    <td width="50%">
+                      <strong>{this.props.statVarId}</strong>
+                    </td>
+                  </tr>
+                  {this.sortedDates.map((date) => {
+                    if (date in this.props.sourceSeries.val) {
+                      return (
+                        <tr
+                          className={obsTableRowClass}
+                          key={date}
+                          onClick={() => this.redirectToObsPage(date)}
+                        >
+                          <td width="50%">{date}</td>
```

Comment:
it could be 40/60 (or another ratio) since the dates will be fixed length, but stat var names pretty long..

Reply:
done